dev_guide/builds.adoc

```diff
@@ -549,156 +594,425 @@ source:
-When defining a `*BuildConfig*`, you can define triggers to control the
-circumstances in which the `*BuildConfig*` should be run. The following build
-triggers are available:
+. Add a `*sourceSecret*` field into the `*source*` section inside the
+`*BuildConfig*` and set it to the name of the `*secret*` that you created.
+In this case `*sshsecret*`:
++
+====
 
-* link:#webhook-triggers[Webhook]
-* link:#image-change-triggers[Image change]
-* link:#config-change-triggers[Configuration change]
+----
+{
+  "apiVersion": "v1",
+  "kind": "BuildConfig",
+  "metadata": {
+    "name": "sample-build",
+  },
+  "parameters": {
+    "output": {
+      "to": {
+        "kind": "ImageStreamTag"
+        "name": "sample-image:latest"
+      }
+    },
+    "source": {
+      "git": {
+        "uri": "git@repository.com:user/app.git" <1>
+      },
+      "sourceSecret": {
+        "name": "sshsecret"
+      },
+      "type": "Git"
+    },
+    "strategy": {
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "python-33-centos7:latest"
+        }
+      },
+      "type": "Source"
+    }
+  }
+----
+<1> The URL of private repository, accessed by a private SSH key, is usually
+in the form `git@example.com:<username>/<repository>.git`.
+====
+
+[[other-authentication]]
+==== Other
+
+In case the cloning of your application is dependent on a `CA certificate`, `.gitconfig`
+file or both, you can create a secret that contains them, add it to the builder service
+account and then your `BuildConfig`.
+
+. Create desired type of `*secret*`:
+
+.. To create a secret from a `.gitconfig`:
++
+====
+----
+$ oc secrets new mysecret .gitconfig=path/to/.gitconfig
+----
+====
+.. To create a secret from a `CA certificate`:
++
+====
+----
+$ oc secrets new mysecret ca.crt=path/to/certificate
+----
+====
+.. To create a secret from a `CA certificate` and `.gitconfig`:
++
+====
+----
+$ oc secrets new mysecret ca.crt=path/to/certificate .gitconfig=path/to/.gitconfig
+----
+====
+
+[NOTE]
+====
+Please note that SSL verification can be turned off, if `sslVerify=false` is set
+for the `http` section in your `.gitconfig` file:
+----
+[http]
+        sslVerify=false
+----
+====
+
+. Add the `*secret*` to the builder service account:
++
+====
+----
+$ oc secrets add serviceaccount/builder secrets/mysecret
+----
+====
+
+
+== Dockerfile Source
+
+Questions:
+
+1. Is it only allowed when `strategy: "Docker"`? Setting `strategy: "Source"`
+fails validation, but `strategy: "Custom"` passes. `oc new-build` only allow
+`--strategy=source`.
+
+2. Can there be two build sources at once? [This
+comment](https://github.com/openshift/origin/pull/4717/files#diff-43dd40b4c2b85b388a3d9116729032e4R123)
+suggests so.
+```
+{
+  source: {
+    type: "Git",
+    git: {...},
+    dockerfile: "..."
+  }
+}
+```
+
+3. If it is valid to have `source.git` and `source.dockerfile` set at the same
+time, does it imply that the `type` must be `"Git"`?
```

Comment:
i'm working on a pull to nuke it from the internal api, with the intent to ultimately remove it from the public api in v2 or whatever version allows us to.

i'm also going to remove it from BuildStrategy.